api.js

```diff
@@ -0,0 +1,164 @@
+"use strict";
+
+const {classes: Cc, interfaces: Ci, results: Cr, utils: Cu} = Components;
+
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+
+Cu.import("resource://devtools/shared/event-emitter.js");
+Cu.import("resource://gre/modules/ExtensionUtils.jsm");
+const {SingletonEventManager} = ExtensionUtils;
+
+XPCOMUtils.defineLazyModuleGetter(this, "webrtcUI",
+                                  "resource:///modules/webrtcUI.jsm");
+
+let origReceiveMessage = null;
```

Comment:
The underlying webrtcui.jsm makes this necessary, but doing a monkey patch like this is quite brittle.  It feels like webrtcui should be fixed prior to moving this api beyond experiment phase.  Maybe @Standard8 could give some input on this issue. https://dxr.mozilla.org/mozilla-central/source/browser/modules/webrtcUI.jsm#177